marketplace-service/src/main/java/ch/akros/marketplace/service/service/TopicService.java

```diff
@@ -214,50 +219,107 @@ public void deleteTopic(Long topicId) {
   }
 
   public TopicSearchListResponseDTO searchTopic(TopicSearchRequestDTO topicSearchRequestDTO) {
-    TopicSearchListResponseDTO result = new TopicSearchListResponseDTO();
-
-    Topic topic = new Topic();
-    topic.setRequestOrOffer(topicSearchRequestDTO.getRequestOrOffer());
-    if (topicSearchRequestDTO.getSearchValues() != null) {
-      List<TopicValue> collect = topicSearchRequestDTO.getSearchValues().stream().map(value -> {
-        return TopicValue.builder().value(value.getValue()).build();
-      }).collect(Collectors.toList());
+        TopicSearchListResponseDTO result = new TopicSearchListResponseDTO();
+
+        List<TopicValue> topicValues = Collections.emptyList();
+        if (topicSearchRequestDTO.getSearchValues() != null) {
+            topicValues = topicSearchRequestDTO.getSearchValues().stream()
+                    .map(value -> TopicValue.builder()
+                            .value(value.getValue())
+                            .topicValueId(value.getFieldId())
+                            .build())
+                    .collect(Collectors.toList());
+        }
+
+        Topic topic = Topic.builder()
+                .requestOrOffer(topicSearchRequestDTO.getRequestOrOffer())
+                .subCategory(SubCategory.builder()
+                        .subCategoryId(topicSearchRequestDTO.getSubcategoryId())
+                        .build())
+                .build();
+
+        Example<Topic> exampleTopic = Example.of(topic, ExampleMatcher.matchingAll());
+        List<TopicSearchResponseDTO> topicList = topicRepository.findAll(exampleTopic).stream()
+                .map(this::toTopicSearchResponseDTO)
+                .collect(Collectors.toList());
+
+        for (TopicValue topicValue : topicValues) {
+            int FROM_SIZE = 25;
```

Comment:
If they are not modified, you can declare FROM_SIZE and TO_PRICE as final at the top of the file.

Also, it would be nice to have comments at least for the public methods, it sometimes get difficult to understand what the code is doing.

Reply:
By resolving warnings before push, I by mistake  converted these two constants to a local variable. Now they are on the top of the file. 
 
 About comments. As in the whole class there was no comments for public methods, I also haven´t written any.  May be we should start to do it.